RUFAS/input_manager.py

```diff
@@ -780,6 +780,54 @@ def get_metadata(self, metadata_address: str) -> Any:
             raise KeyError(f"Data not found: Cannot find \"{metadata_address}\", "
                            f"\"{parent_address}\" does not have attribute \"{invalid_key}\".")
 
+    def get_data_keys_by_properties(self, target_properties: str) -> list[str]:
+        """
+        Retrieves list of keys that point to data which have specified properties.
+
+        Parameters
+        ----------
+        target_properties : str
+            The name of the property that is being searched for.
+
+        Returns
+        -------
+        list[str]
+            List of keys that point to data in the Input Manager which adhere to the target property.
```

Comment:
I'm not sure how to make this clearer but it feels like we're missing some information. It's admitedly a very complicated and specific thing. I'm just throwing out some ideas here to try and encompass more of what I think should be in here for the user to understand what's returned by this function.

```suggestion
            List of keys that point to the data within the Input Manager's pool which adhere to the metadata's guidelines for the target properties.
```

Reply:
Your suggested change is good - it's much clearer that we are looking for keys that will be used to get information from the metadata. I will update this docstring, but will probably need more suggestions on it.